tests/test_io_dummy_param.rs

```diff
@@ -2,59 +2,100 @@
         },
         utils::init_tracing,
     };
     use keccak_asm::Keccak256;
     use num_bigint::BigUint;
+    use rand::Rng;
     use std::sync::Arc;
     use tracing::info;
 
     const SIGMA: f64 = 4.578;
 
     #[test]
+    #[allow(clippy::needless_range_loop)]
     fn test_io_just_mul_enc_and_bit() {
         init_tracing();
         let start_time = std::time::Instant::now();
         let params = DCRTPolyParams::new(4, 2, 17);
         let log_q = params.modulus_bits();
         let switched_modulus = Arc::new(BigUint::from(1u32));
+
+        let sampler_uniform = DCRTPolyUniformSampler::new();
+        let sampler_hash = DCRTPolyHashSampler::<Keccak256>::new([0; 32]);
+        let sampler_trapdoor = DCRTPolyTrapdoorSampler::new(&params, SIGMA);
+
+        // 1. Generate RLWE ciphertext (a, b) for input k
+        // b = a * t - k * q/2 + e
+        let rlwe_encryption_sigma = 3.0;
+
+        // Generate random plaintext bits
+        let k = sampler_uniform.sample_poly(&params, &DistType::BitDist);
+
+        // Encrypt the plaintext
+        let (a, b, t) = encrypt_rlwe(&params, &sampler_uniform, rlwe_encryption_sigma, &k);
+
+        // decompose the polynomials a and b into bits
+        let a_bits = a.decompose(&params);
+        let b_bits = b.decompose(&params);
+
+        assert!(a_bits.len() == b_bits.len());
+        assert!(a_bits.len() == log_q);
+
+        // 2. Create a public circuit
+        // Input: a_bits[0], ..., a_bits[logq - 1], b_bits[0], ..., b_bits[logq - 1], x
+        // Output: a_bits[0] AND x, ..., a_bits[logq - 1] AND x, b_bits[0] AND x, ..., b_bits[logq -
+        // 1] AND x
+        // a_bits and b_bits are hardcoded inside the circuit
         let mut public_circuit = PolyCircuit::new();
-        {
-            let inputs = public_circuit.input(log_q + 1);
-            let mut outputs = vec![];
-            let eval_input = inputs[log_q];
-            for enc_input in inputs[0..log_q].iter() {
-                let muled = public_circuit.and_gate(*enc_input, eval_input);
-                outputs.push(muled);
-            }
-            public_circuit.output(outputs);
+        let x_id = public_circuit.input(1)[0];
+
+        let mut public_circuit_outputs = Vec::new();
+
+        let a_bits_vecs = a_bits.iter().map(|poly| poly.to_bool_vec()).collect::<Vec<_>>();
+        let b_bits_vecs = b_bits.iter().map(|poly| poly.to_bool_vec()).collect::<Vec<_>>();
+
+        for i in 0..a_bits_vecs.len() {
+            let a_ith_bits_const = public_circuit.const_bit_poly(&a_bits_vecs[i]);
+            let a_ith_bits_and_x = public_circuit.and_gate(a_ith_bits_const, x_id);
+            public_circuit_outputs.push(a_ith_bits_and_x);
         }
 
+        for i in 0..b_bits_vecs.len() {
+            let b_ith_bits_const = public_circuit.const_bit_poly(&b_bits_vecs[i]);
+            let b_ith_bits_and_x = public_circuit.and_gate(b_ith_bits_const, x_id);
+            public_circuit_outputs.push(b_ith_bits_and_x);
```

Comment:
I think this embedding should be performed in the `obfuscate` function.
The public circuit only assumes that it can receive `a_bits_vecs ` and `b_bits_vec` as the first `2*log_q` inputs.

Reply:
My point is that `a_bits_vecs` and `b_bits_vecs` should be hardcoded inside the circuit and not passed as inputs

Reply:
> My point is that a_bits_vecs and b_bits_vecs should be hardcoded inside the circuit and not passed as inputs

The original implementation already hardcoded them inside the circuit.
Specifically, it calls the given public circuit which receive a_bits_vecs  and b_bits_vec as the first 2*log_q inputs as the sub-circuit, and the main circuit finally defines these bits as hardcoded wires.
Therefore, these bits are not input to the final built circuit.
https://github.com/MachinaIO/diamond-io/blob/main/src/io/utils.rs#L104-L124